### PR TITLE
Support soft line break.

### PR DESCRIPTION
**Pull request by YeungKC**
The markdown spec doesn't support soft line breaks, but many, such as GitHub, do, and we should provide that option.

close https://github.com/flutter/flutter/issues/88271

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated `CHANGELOG.md` to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the `#hackers-new` channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: packages/flutter_markdown/test/text_test.dart [hunk 1]
```diff
@@ -136,6 +136,26 @@ void defineTests() {
         expectTextStrings(widgets, <String>['line 1. line 2.']);
       },
     );
+
+    testWidgets(
+      'soft line break',
+          (WidgetTester tester) async {
```
Comment (domesticmouse) [inline packages/flutter_markdown/test/text_test.dart:142]: Please see the test output, it looks like this line is failing due to format.

File: packages/flutter_markdown/test/text_test.dart [hunk 2]
```diff
@@ -136,6 +136,26 @@ void defineTests() {
...
+              data: data,
+              softLineBreakPattern: true,
+            ),
+          ),
+        );
+
+        final Iterable<Widget> widgets = tester.allWidgets;
+        expectWidgetTypes(widgets,
+            <Type>[Directionality, MarkdownBody, Column, Wrap, RichText]);
+        expectTextStrings(widgets, <String>['line 1.\nline 2.']);
```
Comment (domesticmouse) [inline packages/flutter_markdown/test/text_test.dart:156]: What would the output be if `softLineBreakPattern` was set to false?

File: packages/flutter_markdown/lib/src/builder.dart [hunk 3]
```diff
@@ -145,6 +146,14 @@ class MarkdownBuilder implements md.NodeVisitor {
   /// Default tap handler used when [selectable] is set to true
   final VoidCallback? onTapText;
 
+  /// The soft line break pattern is used to identify the spaces at the end of a
+  /// line of text and the leading spaces in the immediately following the line
+  /// of text.
+  ///
+  /// Default these spaces are removed in accordance with the Markdown
+  /// specification on soft line breaks when lines of text are joined.
+  final bool softLineBreakPattern;
```
Comment (domesticmouse) [inline packages/flutter_markdown/lib/src/builder.dart:155]: This argument name doesn't make any sense. A `bool` that is the soft line break pattern?

File: packages/flutter_markdown/test/text_test.dart (hunk 2 above)
Comment (YeungKC) [reply packages/flutter_markdown/test/text_test.dart:157]: Same style as current master, no change

File: packages/flutter_markdown/lib/src/builder.dart (hunk 3 above)
Comment (YeungKC) [reply packages/flutter_markdown/lib/src/builder.dart]: What do you think about name change to `softLineBreak`?

File: packages/flutter_markdown/lib/src/widget.dart [hunk 4]
```diff
@@ -247,6 +248,14 @@ abstract class MarkdownWidget extends StatefulWidget {
   /// does not allow for intrinsic height measurements.
   final MarkdownListItemCrossAxisAlignment listItemCrossAxisAlignment;
 
+  /// The soft line break pattern is used to identify the spaces at the end of a
```
Comment (domesticmouse) [inline packages/flutter_markdown/lib/src/widget.dart:251]: Please update the doc string

**Review by domesticmouse — APPROVED** [review]
Lgtm once the docstring is updated

File: packages/flutter_markdown/lib/src/builder.dart (hunk 3 above)
Comment (domesticmouse) [reply packages/flutter_markdown/lib/src/builder.dart]: Sounds good 👍

**Comment by antonio-gravity** [comment]
Great! i cant wait for this

**Comment by marta-blazquez** [comment]
Hi! I am not able to get any result using: "\ n, however, IT" \ "is working in some circumstances; for example:

**Review by domesticmouse — APPROVED** [review]

**Comment by domesticmouse** [comment]
> Hi! I am not able to get any result using: "\ n" , however, this character " \ " is working in some circumstances; for example:
> 
> * After a link, and before a text.
> * After a text and before a picture.
> 
> Finally to add some extra space between a link and a title I have to use a empty code block.

Hey @marta-blazquez, can you raise a bug request on the [flutter/flutter Issue tracker](https://github.com/flutter/flutter/issues/new), and tag it with the label `p: flutter_markdown`.

**Pull request merged**